crates/daphne-worker/src/storage_proxy/mod.rs

```diff
@@ -223,12 +223,24 @@ async fn handle_kv_request(ctx: &mut RequestContext<'_>, key: &str) -> worker::R
             }
         }
         worker::Method::Post => {
+            let expiration_header = ctx.req.headers().get("X-Expiration")?;
+            let expiration_unix_timestamp = if let Some(expiration) = expiration_header {
+                expiration.parse::<u64>().map_err(|e| {
+                    worker::Error::RustError(format!("Failed to parse expiration header: {e:?}"))
+                })
+            } else {
+                Ok(0)
+            }?;
```

Comment:
instead of using a magic 0 value, you can use option
```suggestion
            let expiration_unix_timestamp = expiration_header.map(|ex| ex.parse()).transpose().map_err(|e| {
                worker::Error::RustError(format!("Failed to parse expiration header: {e:?}"))
            })?;
```

if transpose is too cryptic (I sometimes find it a bit cryptic) you can do this instead.
```suggestion
            let expiration_unix_timestamp = if let Some(expiration) = expiration_header {
                Some(expiration.parse::<u64>().map_err(|e| {
                    worker::Error::RustError(format!("Failed to parse expiration header: {e:?}"))
                })?)
            } else {
                None
            }?;
```

Reply:
Thanks I didn't know about transpose. Pretty useful